hashes/Cargo.toml

```diff
@@ -13,13 +13,10 @@ edition = "2018"
 exclude = ["tests", "contrib"]
 
 [features]
-default = ["std"]
-std = ["internals/alloc"]
-schemars = ["actual-schemars", "dyn-clone"]
-# If you disable std, you can still use a Write trait via the core2 feature.
-# You can also use ToHex via the alloc feature, as it requires Vec/String.
-# And you can still just disable std by disabling default features, without enabling these two.
+default = []
```

Comment:
To be clear -- you are suggesting we have `default = ["std"]` for hashes?

I agree, but I don't think it's clear from your earlier comments here.

Reply:
Yes, that's what I meant. "nothing else should depend on it" doesn't apply to anything that doesn't have `-internals`, `-private` or a similar suffix. :)

Reply:
Done, excuse me for being slow but I don't understand the policy for when to use "std" as default, or if its not policy what's the technical reasoning. If anyone has time please tell me [like I'm 5]

Reply:
We're trying to optimize "use the library in most projects without too much hassle". For downstream *libraries* having default features is a hassle (assuming those libraries don't necessarily need them). For most downstream *binaries* not having `std` by default is hassle since they probably use `std` anyway. For embedded projects `std` is hassle.

We assume that most downstream projects are binaries.

`bitcoin`, `bitcoin_hashes`, `secp256k1`, `hex`, `bech32` and others have maintained public API, intended to be used by any downstream project, so making `std` the default makes sense. `bitcoin-internals` have zero downstream binaries (apart from potential fools using it but we aren't obligated to support them) and is only used by our *libraries* so adding defaults is a hassle for us and doesn't help anyone (except the fools).

Reply:
Awesome, got it. Thanks man